src/NuGet.Indexing/ExpandAcronymsFilter.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using Lucene.Net.Analysis;
+using Lucene.Net.Analysis.Tokenattributes;
+
+namespace NuGet.Indexing
+{
+    public sealed class ExpandAcronymsFilter : TokenFilter
+    {
+        private readonly IAcronymExpansionProvider _acronymExpansionProvider;
+
+        private readonly ITermAttribute _termAttribute;
+        private readonly IPositionIncrementAttribute _positionIncrementAttribute;
+        private readonly HashSet<string> _tokenSet;
+        private State _currentState;
+
+        public ExpandAcronymsFilter(TokenStream input, IAcronymExpansionProvider acronymExpansionProvider) 
+            : base(input)
+        {
+            _acronymExpansionProvider = acronymExpansionProvider;
+
+            _termAttribute = AddAttribute<ITermAttribute>();
+            _positionIncrementAttribute = AddAttribute<IPositionIncrementAttribute>();
+            _tokenSet = new HashSet<string>();
+        }
+
+        public override bool IncrementToken()
+        {
+            if (_tokenSet.Count > 0)
+            {
+                RestoreState(_currentState);
+
+                var tokenAtHead = _tokenSet.First();
+
+                _termAttribute.SetTermBuffer(tokenAtHead);
+                _tokenSet.Remove(tokenAtHead);
+
+                _positionIncrementAttribute.PositionIncrement = 0;
+
+                return true;
+            }
+
+            try
+            {
+                if (!input.IncrementToken()) // end of stream; no more tokens on input stream
+                {
+                    return false;
+                }
+            }
+            catch (Exception ex)
```

Comment:
What exception could be thrown here and why? (is there a more specific exception we can catch instead?)